spinalcordtoolbox/reports/qc.py

```diff
@@ -157,7 +203,8 @@ def label_vertebrae(self, mask, ax):
                 a.append(val)
                 index = int(val)
                 if index in self._labels_regions.values():
-                    color = self._labels_color[index]
+                    # NB: We need to subtract `min` to convert the label value into an index for the color list
+                    color = color_list[index - labels.min()]
                     y, x = center_of_mass(np.where(data == val, data, 0))
                     # Draw text with a shadow
                     x += data.shape[1] / 25
```

Comment:
Originally asked by @yw7:

> Work great for me (: Thank you!!!!
> 
> ![image](https://private-user-images.githubusercontent.com/36595323/308342841-8e1ec2e0-4c49-41a5-a80f-8f66602d5639.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxNDQ0OTYsIm5iZiI6MTcwOTE0NDE5NiwicGF0aCI6Ii8zNjU5NTMyMy8zMDgzNDI4NDEtOGUxZWMyZTAtNGM0OS00MWE1LWE4MGYtOGY2NjYwMmQ1NjM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI4VDE4MTYzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3ODM5MDdiMjgyN2EwZTYwZjgyMGQ5ZGFjNjRjMDg2YTg0NjY5MTE3MGY4YzJiN2VlMTkzMGNjNmU4ZDlhZTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7hSYXulXx2ZwgflubPZ3vvTURLf1u1BNOLUTzU8a3RE)
> 
> About this, will it be possible to remove the labels?

I think that the way you suggest (adding a different option to `-p`) would be very doable. :)
   -

Reply:
I was also thinking of perhaps some sort of automatic way of disabling the label text under certain circumstances. We currently add text for these labels:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/e3d00c58cbe98e615e469c63edcd6693f3a0004d/spinalcordtoolbox/reports/qc.py#L42-L48

I was thinking of something like:

- If there are extra labels beyond these hardcoded labels, then don't add the text.
- If the text would overlap onto a different segment (e.g. the vertebrae segments), then don't add the text.

But, the more I think about it, the more I think that this wouldn't be foolproof (and may conflict with user desires). So, I think I agree with the suggestion of making it an explicit option.

Reply:
> Work perfect for me!

Glad to hear!

> Also, not sure if for now and here.. but I thought you can add something like this to overwrite _labels_regions:
-text-labels L5:18,L4:19,L3:20,L2:21,L1:22,T12:23,T11:24,T10:25,T9:26,T8:27,T7:28,T6:29,T5:30,T4:31,T3:32,T2:33,T1:34,C7:35,C6:36,C5:37,C4:38,C3:39,C2:40,C1:41,S:92,SC:200,CSF:201,L5S:202,L4L5:203,L3L4:204,L2L3:205,L1L2:206,T12L1:207,T11T12:208,T10T11:209,T9T10:210,T8T9:211,T7T8:212,T6T7:213,T5T6:214,T4T5:215,T3T4:216,T2T3:217,T1T2:218,C7T1:219,C6C7:220,C5C6:221,C4C5:222,C3C4:223,C2C3:224

Just to clarify -- do you mean that you would like this feature? Or, do you mean that the name "-text-labels" is a bit confusing, since the name suggests that you might be able to pass the "text labels" that you want to be displayed?

Would something like "-show-text-labels" be clearer?

Reply:
Actually Both. But I thought we can do both in one param so if it will be pass with 0 it will remove the labels and if pass with a dictionary it will overwrite existing labels.
Also we can leave it as is for now and open another issue for later to add functionality to this param to overwrite the default labels.